torch/distributed/pipelining/schedules.py

```diff
@@ -3207,3 +3204,24 @@ def _dump_chrometrace(schedule, filename):
 
     with open(filename, "w") as f:
         json.dump({"traceEvents": events}, f)
+
+
+def _check_torch_compile_compatibility(
+    stages: list[_PipelineStageBase], schedule_name: str
+):
+    """
+    Check if the schedule is compatible with torch.compile.
+
+    Args:
+        stages: List of pipeline stages to check
+        schedule_name: Name of the schedule for error message
+
+    Raises:
+        RuntimeError: If any stage uses torch.compile
+    """
+    for stage in stages:
+        if isinstance(stage.submod, OptimizedModule):
```

Comment:
This should check all submodules right? Otherwise use cases like in TorchTitan where transformer blocks are compiled individually instead of the root module are not covered by this check?